environment.go

```diff
@@ -3,48 +3,111 @@ package gantry
 import (
 	"bufio"
 	"bytes"
+	"errors"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
+	"path/filepath"
 	"strings"
 	"text/template"
 
 	"github.com/ad-freiburg/gantry/types"
 	"github.com/ghodss/yaml"
 )
 
-// PipelineEnvironment stores additional data for pipelines.
+// PipelineEnvironment stores additional data for pipelines and steps.
 type PipelineEnvironment struct {
+	Version     string                  `json:"version"`
 	Machines    []Machine               `json:"machines"`
 	LogSettings LogSettings             `json:"log"`
 	Environment types.MappingWithEquals `json:"environment"`
 	TempDirPath string                  `json:"tempdir"`
+	Services    ServiceMetaList         `json:"services"`
+	Steps       ServiceMetaList         `json:"steps"`
 	tempFiles   []string
 	tempPaths   map[string]string
 }
 
-func NewPipelineEnvironment() *PipelineEnvironment {
-	e := &PipelineEnvironment{}
-	e.tempPaths = make(map[string]string, 0)
-	return e
-}
+// NewPipelineEnvironment builds a new environment merging the current
+// environment, the environment given by path and the user provided steps to
+// ignore.
+func NewPipelineEnvironment(path string, environment types.MappingWithEquals, ignoredSteps types.StringSet) (*PipelineEnvironment, error) {
+	// Set defaults
+	e := &PipelineEnvironment{
+		tempPaths:   make(map[string]string, 0),
+		Environment: types.MappingWithEquals{},
+	}
+	e.updateEnvironment(environment)
+	e.updateIgnoredSteps(ignoredSteps)
 
-func (e *PipelineEnvironment) Load(path string) error {
-	if _, err := os.Stat(GantryEnv); path == "" && os.IsExist(err) {
-		path = GantryEnv
+	// Import settings from file
+	dir, err := os.Getwd()
+	if err != nil {
+		return e, err
+	}
+	defaultPath := filepath.Join(dir, GantryEnv)
+	if _, err := os.Stat(defaultPath); path == "" && err == nil {
+		path = defaultPath
 	}
 	file, err := os.Open(path)
 	if err != nil {
-		return err
+		return e, err
 	}
 	defer file.Close()
 
 	data, err := ioutil.ReadAll(file)
 	if err != nil {
-		return err
+		return e, err
+	}
+	// TODO(lehmann): apply current environment to .env.yml ?
```

Comment:
Is this TODO still relevant?

Reply:
It's semi-relevant as I haven't spent more time thinking about it. But I'll remove it for now, if needed this can be opened as an issue.